torch/csrc/api/src/nn/modules/embedding.cpp

```diff
@@ -83,6 +83,15 @@ EmbeddingBagImpl::EmbeddingBagImpl(const EmbeddingBagOptions& options_) : option
 }
 
 void EmbeddingBagImpl::reset() {
+  if (options.padding_idx() != c10::nullopt) {
+    if (*options.padding_idx() > 0) {
```

Comment:
`options->padding_idx()` isn't the same as `*options.padding_idx()`.

`options->padding_idx()` is equivalent to `(*options).padding_idx()`. Since `options` is not a pointer and doesn't have an dereference operator override, it fails to compile.

`*options.padding_idx()`, on the other hand, is equivalent to `*(options.padding_idx())`. So it's dereferencing the `c10::optional<int64_t>` returned by `padding_idx()`, which has a dereference operator override. Another equivalent is `options.padding_idx().value()`, which I think is more readable, so I'll use that instead

Reply:
I'll also change `options.padding_idx() != c10::nullopt` to `options.padding_idx().has_value()`--I think that's more readable as well.

Reply:
Indeed, it slipped my mind that `.` has a higher precedence than `*`. I agree with `options.padding_idx().value()` being more readable.